banyand/liaison/grpc/discovery.go

```diff
@@ -257,3 +295,73 @@ func (e *entityRepo) getLocator(id identity) (partition.EntityLocator, bool) {
 	}
 	return el, true
 }
+
+var _ schema.EventHandler = (*shardingKeyRepo)(nil)
+
+type shardingKeyRepo struct {
```

Comment:
In my view, the `shardingKeyRepo` serves a similar purpose to the `entityRepo`. Instead of creating a new `shardingKeyRepo`, why not modify the existing `entityRepo` to accept the `shardingKey` as the locator, provided that the `shardingKey` is not empty? This approach would minimize memory overhead compared to adding a separate `shardingKeyRepo`.